### PR TITLE
Implementing new Service Page

### PR DESCRIPTION
**Pull request by lexholden**
- [x] New Service Page
- [x] New reuseable components to share across other pages
  - [x] /layout: new folder with helpful components used across the app
    - [x] ListingTitleLink: Takes a resource/org/service, displays the title, and a popover on hover
    - [x] OrganizationCard: Shows the details of an organization in a card
    - [x] ServiceCard: Service info in a card
  - [x] /listing: new folder with reuseable components for program/org/service pages
    - [x] ActionSidebar - Customizeable action buttons
    - [x] CategoryTag - A small tag that links to a category page
    - [x] RelativeOpeningTime - A small component that turns a schedule into a "Open Now" string
    - [x] TableOfContactInfo - Turns a resource/service/program into a section with contactInfo
    - [x] TableOfOpeningTimes - basic table of schedule info
  - [x] /maps: new folder to consolidate map related components
    - [x] MapOfLocations - plots a small map, all locations have an accordion below with extra info
    - [x] StreetViewImage - Shows a streetview image from an address
  - [x] /ui: Some new Basic UI components, moved Loader/Footer/Navigation here
    - [x] Accordion & AccordionItem - A basic group of collabsible sections that expand with more info on click
    - [x] Datatable - Basic table component with potential for future sorting/filtering enhancements
- [ ] Refactored all existing routable pages into the /pages folder in Root, for clear distinction between entry components and reusable generic ones.
- [x] Refactoring of styles to flatten out redundant files
  - [x] Some more reuseable variables

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -11,14 +11,22 @@ var buildDir = path.resolve(__dirname, 'build');
...
   output: {
     path: buildDir,
     publicPath: '/dist/',
-    filename: 'bundle.js'
+    filename: 'bundle.js',
   },
   resolve: {
     extensions: ['.tsx', '.ts', '.jsx', '.js'],
+    alias: {
+      assets: path.resolve(appRoot, 'assets'),
```
Comment (lexholden) [inline webpack.config.js]: Adding these so we can stop using massive import chains of:

```javascript
// Bad
import { SomeComponent } from '../../../../components/folder/SomeComponent'

// Good
import { SomeComponent } from 'components/folder/SomeComponent'
```

File: package.json [hunk 2]
```diff
@@ -71,13 +71,14 @@
     "testcafe": "^0.18.2",
     "testcafe-browser-provider-saucelabs": "^1.3.0",
     "testcafe-react-selectors": "^1.0.2",
-    "webpack": "^1.13.0",
+    "webpack": "^2.6.1",
```
Comment (lexholden) [inline package.json]: We had webpack twice with 2 different versions

File: package.json [hunk 3]
```diff
@@ -8,43 +8,40 @@
   },
   "main": "app.js",
   "dependencies": {
-    "babel-eslint": "^7.2.3",
```
Comment (lexholden) [inline package.json]: Moved all this to devdeps

File: app/utils/index.js [hunk 4]
```diff
@@ -125,7 +127,9 @@ export function buildHoursText(scheduleDays) {
       if (day.opens_at === 0 && day.closes_at >= 2359) {
         hours = 'Open 24 Hours';
       } else {
-        hours = `Open Now: ${timeToString(day.opens_at)}-${timeToString(day.closes_at)}`;
+        const openStr = timeToString(day.opens_at);
```
Comment (lexholden) [inline app/utils/index.js]: Done this to drop the seconds off the end
```
// Bad
Opens 10:00:00

// Good
Opens 10:00
```

File: app/styles/main.scss [hunk 5]
```diff
@@ -1,6 +1,43 @@
 //////////////////////////////////////////////////////////////////////////////////////////
 // TOP LEVEL IMPORTS - DO NOT ADD CSS HERE
 
-@import 'utils/all';
```
Comment (lexholden) [inline app/styles/main.scss]: We had multiple layers of imports. Feels much cleaner with one entry point that is responsible for importing all major.

File: app/styles/components/_listing-page.scss [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+
+	// &.danger {
+	// 	color: $color-red;
+	// }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+// ORG ASIDE
+
+.listing--aside {
```
Comment (lexholden) [inline app/styles/components/_listing-page.scss]: Will remove the styles for resource after all pages have been converted to .listing styles, with consistent implementation.

File: app/reducers/userReducer.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// import * as types from '../actions/actionTypes';
+
+const initialState = {
```
Comment (lexholden) [inline app/reducers/userReducer.js]: This is mostly used in #460 where I move all location loading into redux

File: app/components/ui/Datatable.jsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Datatable extends React.Component {
```
Comment (lexholden) [inline app/components/ui/Datatable.jsx]: This component is mostly pointless, but should allow us to improve table sorting/rendering functionality consistently down the line.

File: app/components/maps/StreetViewImage.jsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class StreetViewImage extends React.Component {
```
Comment (lexholden) [inline app/components/maps/StreetViewImage.jsx]: Seems we may be moving away from using these at all, but either way here is an easy, reuseable implementation

File: app/components/listing/CategoryTag.jsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Link } from 'react-router';
+
+class CategoryTag extends React.Component {
+  render() {
+    const { category } = this.props;
+    return (
+      <Link className="tag block" to={`/search?query=${category.name}`}>
```
Comment (twolfe2) [inline app/components/listing/CategoryTag.jsx]: Changed this so everything will work with the new search

**Pull request merged**